polars/lazy/functions.ts

```diff
@@ -156,78 +156,109 @@ export function lit(value: any): Expr {
   return _Expr(pli.lit(value));
 }
 
-// // ----------
-// // Helper Functions
-// // ------
+// ----------
+// Helper Functions
+// ------
 
 /**
- * __Create a range expression.__
- * ___
+ * Generate a range of integers.
  *
  * This can be used in a `select`, `with_column` etc.
  * Be sure that the range size is equal to the DataFrame you are collecting.
- * @param low - Lower bound of range.
- * @param high - Upper bound of range.
- * @param step - Step size of the range
- * @param eager - If eager evaluation is `true`, a Series is returned instead of an Expr
+ * @param start - Start of the range (inclusive). Defaults to 0.
+ * @param end - End of the range (exclusive). If set to `None` (default), the value of `start` is used and `start` is set to `0`.
+ * @param step - Step size of the range.
+ * @param dtype - Data type of the range.
+ * @param eager - Evaluate immediately and return a `Series`. If set to `False` (default), return an expression instead.
+ * @returns Expr or Series Column of integer data type `dtype`.
+ * @see {@link intRanges}
  * @example
  * ```
  * > df.lazy()
  * >   .filter(pl.col("foo").lt(pl.intRange(0, 100)))
  * >   .collect()
  * ```
+ * 
+ * 
+ * Generate an index column by using `intRange` in conjunction with :func:`len`.
+ * ```
+ *  df = pl.DataFrame({"a": [1, 3, 5], "b": [2, 4, 6]})
+    df.select(
+    ...     pl.intRange(pl.len()).alias("index"),
+    ...     pl.all(),
+    ... )
+    shape: (3, 3)
+    ┌───────┬─────┬─────┐
+    │ index ┆ a   ┆ b   │
+    │ ---   ┆ --- ┆ --- │
+    │ u32   ┆ i64 ┆ i64 │
+    ╞═══════╪═════╪═════╡
+    │ 0     ┆ 1   ┆ 2   │
+    │ 1     ┆ 3   ┆ 4   │
+    │ 2     ┆ 5   ┆ 6   │
+    └───────┴─────┴─────┘
+ * ```
  */
 export function intRange<T>(opts: {
```

Comment:
the usage of `any` in the other overloads make this so type hinting isn't being picked up properly, I'd suggest changing the overloads as follows:

```ts
export function intRange(opts: {
  start: number | Expr;
  end: number | Expr;
  step: number | Expr;
  dtype: DataType;
  eager?: boolean;
});
/** @deprecated *since 0.15.0* use `start` and `end` instead */
export function intRange(opts: {
  low: number | Expr;
  high: number | Expr;
  step: number | Expr;
  dtype: DataType;
  eager?: boolean;
});
export function intRange(
  start: number | Expr,
  end?: number | Expr,
  step?: number | Expr,
  dtype?: DataType,
  eager?: true,
): Series;
export function intRange(
  start: number | Expr,
  end?: number | Expr,
  step?: number | Expr,
  dtype?: DataType,
  eager?: false,
): Expr;
export function intRange(
  opts: any,
  end?,
  step = 1 as number | Expr,
  dtype = DataType.Int64,
  eager?,
): Series | Expr {
 // implementation
}

Reply:
`__tests__/lazy_functions.test.ts` fails due to:
```
    __tests__/lazy_functions.test.ts:143:27 - error TS2769: No overload matches this call.
      The last overload gave the following error.
        Argument of type '{ low: number; high: number; }' is not assignable to parameter of type '{ low: number | Expr; high: number | Expr; step: number | Expr; dtype: DataType; eager?: boolean | undefined; }'.
          Type '{ low: number; high: number; }' is missing the following properties from type '{ low: number | Expr; high: number | Expr; step: number | Expr; dtype: DataType; eager?: boolean | undefined; }': step, dtype

    143         .gtEq(pl.intRange({ low: 0, high: 3 })),
```